addons/account_payment/models/account_journal.py

```diff
@@ -75,3 +75,14 @@ def _compute_available_payment_method_ids(self):
                     to_remove.append(payment_method.id)
 
                 journal.available_payment_method_ids = [Command.unlink(payment_method) for payment_method in to_remove]
+
+    @api.ondelete(at_uninstall=False)
+    def _unlink_except_linked_to_payment_acquirer(self):
+        linked_acquirers = self.env['payment.acquirer'].sudo().search([]).filtered(
+            lambda acq: acq.journal_id.id in self.ids and acq.state != 'disabled'
+        )
+        if linked_acquirers:
+            raise UserError(_(
+                "You must first deactivate a payment acquirer before deleting its journal.\n"
+                "Linked acquirer(s): %s", ', '.join(acq.display_name for acq in linked_acquirers)
```

Comment:
I only noticed it in this PR, but we should avoid using `(s)` because it's not always possible to translate it correctly.
```suggestion
                "Linked acquirers: %s", ', '.join(acq.display_name for acq in linked_acquirers)
```